hgs-keyvault/azuredeploy.json

```diff
@@ -0,0 +1,451 @@
+                },
+                "vaultCertificates": [
+                  {
+                    "certificateUrl": "[parameters('httpsCertificateUrl')]",
+                    "certificateStore": "My"
+                  },
+                  {
+                    "certificateUrl": "[parameters('encryptionCertificateUrl')]",
+                    "certificateStore": "My"
+                  },
+                  {
+                    "certificateUrl": "[parameters('signingCertificateUrl')]",
+                    "certificateStore": "My"
+                  }
+
+
+                ]
+              }
+            ]
+          },
+          "storageProfile": {
+            "imageReference": {
+              "publisher": "[parameters('imagePublisher')]",
+              "offer": "[parameters('imageOffer')]",
+              "sku": "[parameters('imageSKU')]",
+              "version": "latest"
+            },
+            "osDisk": {
+              "name": "osdisk",
+              "vhd": {
+                "uri": "[concat('http://',variables('storageAccountName'),'.blob.core.windows.net/vhds/','osdisk', copyindex(), '.vhd')]"
+              },
+              "caching": "ReadWrite",
+              "createOption": "FromImage"
+            }
+          },
+          "networkProfile": {
+            "networkInterfaces": [
+              {
+                "id": "[resourceId('Microsoft.Network/networkInterfaces',concat(parameters('nicNamePrefix'),copyindex()))]"
+              }
+            ]
+          },
+          "diagnosticsProfile": {
+            "bootDiagnostics": {
+              "enabled": "true",
+              "storageUri": "[concat('http://',variables('storageAccountName'),'.blob.core.windows.net')]"
+            }
+          }
+        },
+        "resources": [
+          {
+            "name": "DSCInit",
+            "type": "extensions",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2015-06-15",
+            "dependsOn": [
+              "[resourceId('Microsoft.Compute/virtualMachines', concat(parameters('vmNamePrefix'), copyindex()))]"
+            ],
+            "tags": {
+              "displayName": "DSCInit"
+            },
+            "properties": {
+              "publisher": "Microsoft.Compute",
+              "type": "CustomScriptExtension",
+              "typeHandlerVersion": "1.4",
+              "autoUpgradeMinorVersion": true,
+              "settings": {
+
+                "fileUris": [
+                  "https://udayvw.blob.core.windows.net/artefacts/dscinit.ps1"
+                ],
+                "commandToExecute": "[concat('powershell -ExecutionPolicy Unrestricted -File ', variables('DSCInitScriptFileName') , ' -NodeType:' , concat('', copyindex()) ,' -HgsServerPrimaryIPAddress:10.0.0.4 -GenerateSelfSignedCertificate:false',  concat(' -HgsServerPrimaryAdminUsername:', parameters('adminUsername')), concat(' -HgsServerPrimaryAdminPassword:', parameters('adminPassword')) )]"
```

Comment:
this contains a password, so the commandToExecute should go into "protectedSettings" instead of "settings"